### ISSUE TITLE
Change wording of master.key to be inclusive

### ISSUE THREAD

**Issue by jennifersarahpaul** [rails/rails#43601]
### Steps to reproduce
Just spinning up a new app. `$ rails new coolproject`

### Expected behavior
The `config/master.key` file should be renamed to use another term that is inclusive to everyone. Perhaps primary.key or root.key. 

### Actual behavior
The `config/master.key` wording is outdated and rooted in discrimination. Let's make it better!

### System configuration
**Rails version**: 6.0.4.1

**Ruby version**: 2.7.0 and 2.7.3

**Comment by jennifersarahpaul** [comment]
I'd love to tackle this as a contributor (although you Rails folks should decide on the final wording first)

**Comment by jhawthorn** [comment]
👍 I think ` primary.key` is a good idea

**Comment by kaspth** [comment]
I agree. Though let's give it a round or two of trying to find clearer terms while we're here. The purpose of the key is to derive all other keys within a Rails app from it (via `ActiveSupport::KeyGenerator`) — and so there's a huge safety aspect to it too if an attacker gets a hold of it.

Maybe `root_derivation.key` or `generation.key`? I like the former most.

**Comment by JeffMorhous** [comment]
I'm all for making the term clearer while we're making it accessible.  I like `root_derivation.key` or maybe `private_derivation.key`

**Comment by ghousemohamed** [comment]
> 👍 I think ` primary.key` is a good idea

Isn't ` primary.key` also suggestive of there being a ` secondary.key`? I'm completely new to all of this, but I'm inclined towards `main.key`

**Comment by cucho** [comment]
> The config/master.key wording is outdated and rooted in discrimination.

**Master key** is a [concept on its own](https://www.oxfordlearnersdictionaries.com/us/definition/english/master-key?q=master+key) and is not based in the master/slave metaphor.

**Comment by jennifersarahpaul** [comment]
> > The config/master.key wording is outdated and rooted in discrimination.
> 
> **Master key** is a [concept on its own](https://www.oxfordlearnersdictionaries.com/us/definition/english/master-key?q=master+key) and is not based in the master/slave metaphor.


The technical origins are not the only thing that matters anymore. Language is heavily rooted in meaning and that matters and needs to be acknowledged. Regardless of the etymology, the word has associations rooted very heavily in racism and that is relevant. There are alternatives that are inclusive, so why not? 

There are many, many people out there who hear the word “master” and immediately associate with truly awful things before analyzing the technical use case. The word is negative for some and probably neutral for most others at best. Why not make the effort to create a safer space in development for all. 

I’m glad we’re having this discussion. It’s people who are coding after all.

**Comment by oleksii** [comment]
No person is free who is not master of himself...

**Comment by inem** [comment]
Shouldn't the amount of work created by such renames also be taken into consideration?

The amount of people who might be offended by the presence of this in the codebase is very little compared to the amount of people world wide, who don't associate this word with slavery. For them it is just a technical term. And after such rename, all these people will have to:
  1. do renames over all their projects during upgrades
  2. get used to the new term

**Comment by pzelnip** [comment]
> The amount of people who might be offended by the presence of this in the codebase is very little compared to the amount of people world wide, who don't associate this word with slavery. For them it is just a technical term. 

*Strongly* disagree with this reasoning.  The fact that for many it is just a technical term with no empathy or realization that its use is a microaggression *is* the problem efforts like this are trying to address.

Related reading: https://tools.ietf.org/id/draft-knodel-terminology-07.html#name-master-slave

**Comment by inem** [comment]
@pzelnip as engineers we sometimes decide not to solve some a bugs or an issue because cost or effect of solving it is too high compared to the negative effects of the initial issue. I just wanted to bring attention that it could be the case here. 

Of course this is relevant if we're approaching the problem as engineers. And if we do, then the first step would be to agree on methodology to estimate that negative impact of the initial issue.

**Comment by matthewd** [comment]
@inem thanks, but I don't need a spreadsheet to compare the impact of some people feeling excluded vs others being inconvenienced by having to "get used to the new term". Especially if they haven't felt the need to understand any of the context behind the existing word, I'm sure that anyone who needs to interact with it will manage. We rename and move things via deprecation fairly frequently, for a variety of reasons, and everyone seems to keep up.

Backwards compatibility will of course, as always, need to be considered in how the change is implemented, but it's going to happen.

**Comment by pzelnip** [comment]
Is there any update on this issue?  What would it take to have it move forward/get some momentum?

**Comment by p8** [comment]
@pzelnip, for this to get more momentum, I think a PR would help.

About the name...
While ActiveRecord::Encryption uses `primary_key`:
https://github.com/rails/rails/blob/c21c4139e1ceae0289a7ae2cd597b4705a24a71b/activerecord/lib/active_record/encryption/configurable.rb#L21
I think it'll be confusing when we already use `primary_key` for the database column type.

**Comment by rafaelfranca** [comment]
Thank you for the issue but we don't have plans to rename this file.
